zcash_client_backend/src/fees/zip317.rs

```diff
@@ -28,13 +29,17 @@ use super::{
 /// shielded pool and delegates fee calculation to the provided fee rule.
 pub struct SingleOutputChangeStrategy {
     fee_rule: Zip317FeeRule,
+    change_memo: Option<MemoBytes>,
 }
 
 impl SingleOutputChangeStrategy {
     /// Constructs a new [`SingleOutputChangeStrategy`] with the specified ZIP 317
     /// fee parameters.
-    pub fn new(fee_rule: Zip317FeeRule) -> Self {
-        Self { fee_rule }
+    pub fn new(fee_rule: Zip317FeeRule, change_memo: Option<MemoBytes>) -> Self {
+        Self {
+            fee_rule,
+            change_memo,
+        }
     }
```

Comment:
Alternatively, if we wanted to make this a less disruptive API change, we could do:
```suggestion
    pub fn new(fee_rule: Zip317FeeRule) -> Self {
        Self {
            fee_rule,
            None,
        }
    }

    /// Constructs a new [`SingleOutputChangeStrategy`] with the specified ZIP 317
    /// fee parameters and change memo.
    pub fn new_with_memo(fee_rule: Zip317FeeRule, change_memo: Option<MemoBytes>) -> Self {
        Self {
            fee_rule,
            change_memo,
        }
    }
```

Reply:
I prefer the more-disruptive change because it forces people to adapt to the change, and won't then silently do a thing they don't expect.